src/terminal/parser/stateMachine.cpp

```diff
@@ -1270,6 +1437,233 @@ void StateMachine::_EventVt52Param(const wchar_t wch)
     }
 }
 
+// Routine Description:
+// - Processes a character event into an Action that occurs while in the DcsEntry state.
+//   Events in this state will:
+//   1. Execute C0 control characters
+//   2. Ignore Delete characters
+//   3. Store parameter data
+//   4. Collect Intermediate characters
+//   5. Pass through characters
+//   6. If we see a ESC, enter the DcsTermination state.
+//  DCS sequences are structurally almost the same as CSI sequences, just with a
+//      extra data string. It's safe to reuse CSI functions for
+//      determining if a character is a parameter, delimiter, or invalid.
+// Arguments:
+// - wch - Character that triggered the event
+// Return Value:
+// - <none>
+void StateMachine::_EventDcsEntry(const wchar_t wch)
+{
+    _trace.TraceOnEvent(L"DcsEntry");
+    if (_isC0Code(wch))
+    {
+        _ActionExecute(wch);
+    }
+    else if (_isDelete(wch))
+    {
+        _ActionIgnore();
+    }
+    else if (_isCsiInvalid(wch))
+    {
+        _EnterDcsIgnore();
+    }
+    else if (_isNumericParamValue(wch) || _isCsiDelimiter(wch))
+    {
+        _ActionParam(wch);
+        _EnterDcsParam();
+    }
+    else if (_isIntermediate(wch))
+    {
+        _ActionCollect(wch);
+        _EnterDcsIntermediate();
+    }
+    else if (_isDcsPassThroughValid(wch))
+    {
+        _ActionDcsPassThrough(wch);
+        _EnterDcsPassThrough();
+    }
+    else if (_isDcsTerminationInitiator(wch))
+    {
+        _EnterDcsTermination();
+    }
+    else
+    {
+        _EnterGround();
+    }
+}
+
+// Routine Description:
+// - Processes a character event into an Action that occurs while in the DcsIgnore state.
+//   Events in this state will:
+//   1. Enter ground on a String terminator
+//   2. Ignore Delete characters
+//   3. If we see a ESC, enter the DcsTermination state.
+//   4. Ignore everything else.
+// Arguments:
+// - wch - Character that triggered the event
+// Return Value:
+// - <none>
+void StateMachine::_EventDcsIgnore(const wchar_t wch) noexcept
+{
+    _trace.TraceOnEvent(L"DcsIgnore");
+    if (_isStringTerminator(wch))
+    {
+        _EnterGround();
+    }
+    else if (_isDelete(wch))
+    {
+        _ActionIgnore();
+    }
+    else if (_isCsiParamValue(wch) || _isCsiDelimiter(wch) || _isCsiInvalid(wch) || _isIntermediate(wch) || _isDcsPassThroughValid(wch))
+    {
+        _ActionIgnore();
+    }
+    else if (_isDcsTerminationInitiator(wch))
+    {
+        _EnterDcsTermination();
+    }

```

Comment:
Again I don't think we want this here.

Reply:
As I said, the fact that iTerm2 actually supports this transition is making me ponder..

Reply:
I think this is still needed if we want strictly follow that `ESC \` is the same as `ST` in the VT spec [here](https://vt100.net/docs/vt510-rm/chapter4.html).

Reply:
Any time you get an `ESC` (except in the `OscString` and `DcsPassThrough` states), that's going to abort the current sequence and start parsing a new sequence. In the case of `ESC \` , that new sequence will be dispatched via the `ActionEscDispatch` and ultimately ignored, but the outcome is still that the original `DCS` sequence is terminated.

The only time we need to specifically check for `ESC \` is in the pass-through state, because that tells us that the sequence has terminated **successfully**, and we need to actually do something with the data. We don't want to do that when we're in the ignore state.

Reply:
Inside `ActionEscDispatch`, `\` is not recognized as a valid terminator, thus it return `false` to indicate the dispatch is a failure. Is there anything we can and should do about it?

Reply:
That's a good point. Maybe we just need to recognise it as a valid op and return true without doing anything.